isisd/isis_circuit.c

```diff
@@ -267,7 +267,7 @@ void isis_circuit_add_addr(struct isis_circuit *circuit,
 		listnode_add(circuit->ip_addrs, ipv4);
 
 		/* Update Local IP address parameter if MPLS TE is enable */
-		if (circuit->ext && IS_MPLS_TE(circuit->ext)) {
+		if (circuit->ext && IS_EXT_TE(circuit->ext)) {
```

Comment:
Does the comment on line 269 need to be updated too?

Reply:
I believe it's fine, if I understand correctly IS_MPLS_TE is used for areas and IS_EXT_TE for circuits. But maybe @odd22 can chip in and confim?

Reply:
Comment is correct, but proposed correction not. The IS_MPLS_TE macro is the correct one to be used, but not against circuit->ext. It must be:
```
if (circuit->ext && IS_MPLS_TE(circuit->mta)) {
```

circuit-ext structure contains the various TE parameters and the ext->status indicates with assigned bit which TE parameters has been set or not. But, it not serve to determine if MPLS-TE is enable or not. For that purpose, you must used circuit->mta structure and mta->status which indicates it MPLS-TE is enable or not.

Reply:
@odd22 there does not seem to be any field called `mta` in `struct isis_cicuit`: https://github.com/FRRouting/frr/blob/master/isisd/isis_circuit.h#L77
Do you mean the mta field in the area? so `circuit->area->mta` (assuming area is not NULL?)

Reply:
Sorry for the mis typo. I write too fast the review. You are right the correct structure is
`circuit->area->mta`